### PR TITLE
cluster interceptor support

### PR DESCRIPTION
**Pull request by beiwei30**
**What this PR does**:

Refactor ClusterInterceptor to chain mode, and allow to load user defined cluster interceptors.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by gaoxinge** [comment]
Add apache license in new files:
 
- common/extension/cluster_interceptor.go
- cluster/cluster_impl/zone_aware_cluster_interceptor.go
- cluster/cluster_impl/interceptor_invoker.go

to pass license check in ci.

**Comment by beiwei30** [comment]
> Add apache license in new files:
> 
> * common/extension/cluster_interceptor.go
> * cluster/cluster_impl/zone_aware_cluster_interceptor.go
> * cluster/cluster_impl/interceptor_invoker.go
> 
> to pass license check in ci.

sure, will do after I verify the solution.

**Review by gaoxinge — APPROVED** [review]

File: common/extension/cluster_interceptor.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package extension
+
+import "github.com/apache/dubbo-go/cluster"
+
+var (
+	interceptors = make(map[string]func() cluster.Interceptor)
```
Comment (AlexStocks) [inline common/extension/cluster_interceptor.go:23]: 出于防御编程的考虑，建议对其加锁保护

Comment (beiwei30) [reply common/extension/cluster_interceptor.go:25]: done. pls. check again.

**Review by AlexStocks — APPROVED** [review]

**Review by wenxuwan — APPROVED** [review]

**Pull request merged**